sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -299,23 +323,36 @@ static struct snd_soc_dai_link_component dmic_component[] = {
 	}
 };
 
+static struct snd_soc_dai_link_component max98357a_component[] = {
+	{
+		.name = "MX98357A:00",
+		.dai_name = "HiFi",
+	}
+};
+
 static struct snd_soc_dai_link *sof_card_dai_links_create(struct device *dev,
-							  int ssp_port,
+							  int ssp_codec,
+							  int ssp_amp,
 							  int dmic_num,
 							  int hdmi_num)
 {
 	struct snd_soc_dai_link_component *idisp_components;
 	struct snd_soc_dai_link *links;
 	int i, id = 0;
 
-	links = devm_kzalloc(dev, sizeof(struct snd_soc_dai_link) *
-			     (1 + dmic_num + hdmi_num), GFP_KERNEL);
+	if (sof_rt5682_quirk & SOF_SPEAKER_AMP_PRESENT) {
+		links = devm_kzalloc(dev, sizeof(struct snd_soc_dai_link) *
+				(2 + dmic_num + hdmi_num), GFP_KERNEL);
+	} else {
+		links = devm_kzalloc(dev, sizeof(struct snd_soc_dai_link) *
+				(1 + dmic_num + hdmi_num), GFP_KERNEL);
```

Comment:
can we simplify, we have two cases where we compute the number of links in this file. One should be enough, let's try and factor all this.